test/new-e2e/pkg/utils/e2e/client/agent_client.go

```diff
@@ -161,10 +149,41 @@ func makeStatusEndpointCommand(params *agentclientparams.Params, osFamily osComp
 	}
 
 	statusEndpoint := fmt.Sprintf(url, port)
-	return curlCommand(statusEndpoint, params.AuthToken), true, nil
+	return curlCommand(statusEndpoint, params.AuthToken, osFamily), true, nil
+}
+
+func ensureAuthToken(params *agentclientparams.Params, osFamily osComp.Family, host *Host) error {
+	if params.AuthToken != "" {
+		return nil
+	}
+
+	getAuthTokenCmd := fetchAuthTokenCommand(params.AuthTokenPath, osFamily)
+	authToken, err := host.Execute(getAuthTokenCmd)
+	if err != nil {
+		return fmt.Errorf("could not read auth token file: %v", err)
+	}
+	params.AuthToken = strings.TrimSpace(authToken)
+
+	return nil
+}
+
+func fetchAuthTokenCommand(authTokenPath string, osFamily osComp.Family) string {
+	if osFamily == osComp.WindowsFamily {
+		return fmt.Sprintf("Get-Content -Raw -Path %s", authTokenPath)
+	}
+
+	return fmt.Sprintf("sudo cat %s", authTokenPath)
 }
 
-func curlCommand(endpoint string, authToken string) string {
+func curlCommand(endpoint string, authToken string, osFamily osComp.Family) string {
```

Comment:
The function name no longer match the body. Since depending on the family we use `curl` or `Invoke-WebRequest`. Could we rename the function to something like: `fetchStatus` or something like that 😄

Reply:
Done 49c9c75c3d1fa02405337ee244245f55bd01dbaa